test/OpenTelemetry.Tests/Implementation/Trace/ActivityExtensionsTest.cs

```diff
@@ -0,0 +1,72 @@
+﻿// <copyright file="ActivityExtensionsTest.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Diagnostics;
+using OpenTelemetry.Trace.Configuration;
+using Xunit;
+
+namespace OpenTelemetry.Trace.Test
+{
+    public class ActivityExtensionsTest
+    {
+        private const string ActivitySourceName = "test.status";
+        private const string ActivityName = "Test Activity";
+
+        [Fact]
+        public void SetStatus()
+        {
+            using var openTelemetrySdk = OpenTelemetrySdk.EnableOpenTelemetry(b => b
+                                                   .AddActivitySource(ActivitySourceName));
+
+            using var source = new ActivitySource(ActivitySourceName);
+            using var activity = source.StartActivity(ActivityName);
+            activity.SetStatus(Status.Ok);
```

Comment:
Might need another test case to make sure the last call wins:

```csharp
activity.SetStatus(Status.Cancelled);
activity.SetStatus(Status.Ok);

Assert.True(activity.GetStatus().IsOk);
```

Reply:
`Activity` does not provide an option to update or delete `Tags` entry.  If we want to achieve this functionality we need to add the key `ot.status_code` more than once to the activity tags and read the last entry. This may not be the best option for consumers.

Reply:
With the upcoming changes in .NET Activity, we'll be using activity.SetTag, instead of AddTag. At that time, we'll need this test and feature

Reply:
Added a new test with skip attribute, we could revisit later to change AddTag to SetTag.